testing/btest/Baseline/core.ip-broken-header/weird.log

```diff
@@ -89,7 +89,6 @@
 1500557631.000000	-	9c00:7265:6374:6929:600:0:6904:ff	0	3bbf:ff00:40:0:ffff:ffff:fffd:f7ff	0	invalid_inner_IP_version	-	F	zeek
 1500557631.000000	-	9c00:7463:2a72:6929:400:0:6904:ff	0	3bbf:ff00:40:0:ffff:ffff:fffd:f7ff	0	invalid_inner_IP_version	-	F	zeek
 1500557631.000000	-	b000:7265:6374:6929::8004:ff	0	3bbf:ff80:40:0:ffff:ffff:fffd:f7ff	0	invalid_inner_IP_version	-	F	zeek
-1500557631.000000	-	255.255.0.0	0	255.255.255.237	0	invalid_inner_IP_version	-	F	zeek
```

Comment:
Maybe? The packet that it's complaining about here is this one from $TRACES/trunc/mpls-6in6-broken.pcap.gz:

![Screen Shot 2020-10-09 at 4 08 13 PM](https://user-images.githubusercontent.com/2653616/95638493-ad38c000-0a49-11eb-8b5a-c70add586415.png)

In the old code, `NetSessions::NextPacket` would have gone ahead and parsed out the header, passed it to `DoNextPacket, done a bunch of work on it, and then decided that it had an invalid IP version. It doesn't check the actual value in the header until much later in the process. The ethernet header says it's IPv4, so it must be IPv4 and we can process it as such. Strangely, it reports the weird there as an `invalid_inner_IP_version`, even though it's clearly not an encapsulated packet, but that's completely secondary.

In the new code, one of the first things the IP analyzer does after pulling out the IP header is check that the version is valid. We don't even have an `IP_Hdr` object yet, because we don't know what type it is, even though the Ethernet header assures us that it's IPv4. The header is actually valid in that packet above even though the version is wrong, so the old code was able to pull out orig/resp data. The new code doesn't even try.

As for the reason why this line and the other line are removed from the log, it's because we trim duplicate weirds. Even though these are technically different packets, the weirds are the same and get removed and a counter updates in stats.

Reply:
Thanks for tracking this down, makes sense.  "trim duplicate weirds" explains the missing lines well.